packages/blade-old/src/icons/ArrowRight/ArrowRight.native.js

```diff
@@ -1,30 +1,20 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import Svg, { Path } from 'react-native-svg';
-import { getThemeColors } from '../../_helpers/theme';
-import colors from '../../tokens/colors';
+import { Path } from 'react-native-svg';
+import Icon, { IconPropTypes } from '../../atoms/Icon';
 
-function ArrowRight({ width, height, fill }) {
+function ArrowRight(props) {
```

Comment:
done.

Reply:
@kamleshchandnani found some issues with not passing the props via spreading it on the svg. 

components like Flex, Space are passing down important props to the childrens for styling, but since icons doesn't take those it's messing up the styles. 

This bug is also present in current master, see in AlertCircle we are not spreading the props, but in [Info icon](https://github.com/razorpay/blade/blob/57b1bb99962b8ac8b1b19cb47c4b401cb901483d/packages/blade-old/src/icons/Info/Info.web.js#L5) (which is being used for snapshot testing) does have spreading props. 

Notice how there is no spacing between icons and input text. 

<img width="325" alt="Screenshot 2021-12-01 at 2 07 26 PM" src="https://user-images.githubusercontent.com/35374649/144200207-ef0c68e9-5455-4a64-9b80-de91cff4f65d.png">


After spreading the props in icons. fixed the alignment and spacing bug.
<img width="325" alt="Screenshot 2021-12-01 at 2 09 19 PM" src="https://user-images.githubusercontent.com/35374649/144200509-31351246-a963-4612-9538-b3e93137fbeb.png">

Reply:
oh damn! Need to check because we actually clone the element in flex/space etc and attach props to it. Can you check further if this exactly is the issue?

Reply:
Will check, I'll try passing down className only see if that works. If not we have no choice but to spread all the props it seems 🤔

Reply:
Update: We decided to go with spreading props considering the tradeoffs.